docs/architecture/adr-009.md

```diff
@@ -0,0 +1,89 @@
+As those operations could become more frequent (updating the token mapping) or needs to be trigger by external systems (circuit breaker for gravity module), it would be more practical to define a granular permission system which restrict certain operations to only "some" known addresses.
+
+
+## Terminologies
+
+### Admin
+
+A special address defined in cronos space that can be modified through governance. Usually a multi-sig address.
+
+### Permissioned addresses
+
+An address in Cronos that is allowed to perform a type of operation (message).
+
+### Non-permissioned addresses
+
+An address in Cronos that carry no permissions. 
+
+
+## Decision
+
+### Option 1 : Create a permission system within cronos module for messages.
+
+For the above problem, we propose the following:
+
+- Keep the admin address in Cronos space. The admin has all the permissions and has the ability to change the permission of an address.
+- By default, all addresses in Cronos are non-permissioned
+- Assign to each "restricted" messages in Cronos a permission (integer value) and create in Cronos module a mapping between addresses and permissions that is stored in memory. For now, there are only two messages that require permission : MsgUpdateTokenMapping and MsgTurnBridge.
+- Create a msg type "MsgUpdatePermissions" that only admin can use and allow to update the address permission mapping.
+- Change the logic to always check for the permission before processing the restricted messages.
+
+### Consequences
+
+#### Positive
+
+- Simple implementation self-contained in x/cronos module. Ability to be customized in the future if required.
+- It is still decentralized as the admin address is managed through governance. 
+
+
+#### Negative
+
+- Require some change in the code.
+- Not flexible for potential future change which may require functionalities on the group module.
+
+
+### Option 2 : Use a subset of x/group module functionality to restrict some transactions in cronos
+
+For the above problem, we propose the following:
+
+- For each specific msg, we define a group and for the group, a group policy
+- Use the admin define in cronos space as general admin for the group
+- Set the policy of the group to ``threshold: "1"`` , ``min_execution_period: 0s`` and ``voting_period: 86400s``
+- The policy is associated to a group policy address, each permissioned msg within the proposal needs to be signed by the group policy address (TBD)
+- To be abe to execute a permissioned msg, one member of the group needs to send the msg ``Msg/SubmitProposal``
+
+In Cronos module, we need to restrict ``MsgUpdateTokenMapping`` and ``MsgTurnBridge`` msg to only specific "group policy address".
+
+- Store in the state a mapping between msg and group policy address that are allowed to trigger the msg
+- Only a governance proposal to be able to change the mapping
+- Add code to check the sender address of the message and reject it if it does not belong to one of the address defined in the mapping
+
```

Comment:
It seems the change in x/Cronos is a bit similar with what we need with proposal 1. Not sure if it can be simplified